x-pack/plugins/viz_editor/public/pseudo_plugins/pie_chart_plugin/pie_chart_vis.tsx

```diff
@@ -36,10 +36,14 @@ function sampleVisFunction() {
       },
     },
     context: { types: ['kibana_datatable'] },
-    fn(context: any, args: any) {
-      // normally this would prepare the data, but in this case the kibana_pie function takes care of it
-      // TODO swirtch over to elastic-charts pie as soon as they are ready
-      return { ...context, type: 'kibana_datatable' };
+    fn(context: { rows: any[]; columns: any[] }, args: any) {
+      // we have to reverse the order of the columns because the current kibana pie chart implementation always
+      // expects the value column directly after the bucket column
```

Comment:
Okay so I have a proposal around this: each of these visualizations we're building has specific ordering expectations that are dependent on its editor config. The ordering of columns is not necessarily the order the visualization expects- imagine a split pie chart. Instead, shouldn't we be accessing the data in the columns using some pre-generated accessor? For example, if `count` is the value field, we could map that by id instead of assuming order.

Reply:
Yes, this is definitely better, but it doesn't work for the current pie chart implementation because of `src/legacy/ui/public/agg_response/hierarchical/build_hierarchical_data.js` line 44 - the value column has to be the one after the bucket column. This is just a workaround for the current pie chart, when we have our own implementation (or cleaned up the existing one), it should work as you said. For split pie charts this logic has to be extended here.